core-ui/src/components/Predefined/Details/VirtualServices/HttpRoutes.js

```diff
@@ -0,0 +1,532 @@
+            <>
+              <dd>{t('virtualservices.http-routes.cors.allow-origin')}</dd>
+              <dt>
+                <StringMatch def={origin} />
+              </dt>
+            </>
+          ))}
+          {cors.allowMethods?.map(method => (
+            <>
+              <dd>{t('virtualservices.http-routes.cors.allow-method')}</dd>
+              <dt>{method}</dt>
+            </>
+          ))}
+          {cors.allowHeaders?.map(header => (
+            <>
+              <dd>{t('virtualservices.http-routes.cors.allow-header')}</dd>
+              <dt>{header}</dt>
+            </>
+          ))}
+          {cors.exposeHeaders?.map(header => (
+            <>
+              <dd>{t('virtualservices.http-routes.cors.expose-header')}</dd>
+              <dt>{header}</dt>
+            </>
+          ))}
+          {cors.maxAge && (
+            <>
+              <dd>{t('virtualservices.http-routes.cors.max-age')}</dd>
+              <dt>{cors.maxAge}</dt>
+            </>
+          )}
+          <>
+            <dd>{t('virtualservices.http-routes.cors.allow-credentials')}</dd>
+            <dt>
+              {cors.allowCredentials
+                ? t('common.labels.yes')
+                : t('common.labels.no')}
+            </dt>
+          </>
+        </dl>
+      </LayoutPanel.Body>
+    </LayoutPanel>
+  );
+}
+
+function RedirectPortSelection({ selection }) {
+  const { t } = useTranslation();
+  if (selection === 'FROM_PROTOCOL_DEFAULT') {
+    return t('virtualservices.http-routes.redirect.derive-port-default');
+  } else if (selection === 'FROM_REQUEST_PORT') {
+    return t('virtualservices.http-routes.redirect.derive-port-request');
+  } else {
+    return '';
+  }
+}
+
+function HeaderOperations({ title, operations }) {
+  const { t } = useTranslation();
+  return (
+    <>
+      {title && <div class="header-operation-title">{title}</div>}
+      <ul>
+        {operations.add &&
+          Object.entries(operations.set).map(([header, val]) => (
+            <li>
+              <span className="header-operation">
+                {t('virtualservices.headers.add')}
+              </span>{' '}
+              {header} = {val}
+            </li>
+          ))}
+        {operations.set &&
+          Object.entries(operations.set).map(([header, val]) => (
+            <li>
+              <span className="header-operation">
+                {t('virtualservices.headers.set')}
+              </span>{' '}
+              {header} = {val}
+            </li>
+          ))}
+        {operations.remove &&
+          operations.remove.map(header => (
+            <li>
+              <span className="header-operation">
+                {t('virtualservices.headers.remove')}
+              </span>{' '}
+              {header}
+            </li>
+          ))}
+      </ul>
+    </>
+  );
+}
+
+function HttpMatchRequests({ matches }) {
+  const { t } = useTranslation();
+
+  return (
+    <GenericList
```

Comment:
Do we really need a generic list for that? it seems like a big component for such a small view. why won't we use the same approach as we have for cors policies?

Reply:
Also, the search is still visible and doesn't work.

Reply:
But, with this, you've revealed another tiny bug :)
Please fix this message:
<img width="1302" alt="Screen Shot 2021-12-23 at 13 45 35" src="https://user-images.githubusercontent.com/40663133/147242415-dd52b14d-e132-47ec-9e5d-644cb70fb4e4.png">

Reply:
the idea of using a generic list is because the matches can be more complex, and as such list entries separate matches, while several rules within a match are line-by-line: 
![image](https://user-images.githubusercontent.com/19704958/147440660-21763a00-51ad-4b28-9988-d95cdeb6afb2.png)

Reply:
> 2. Please change this margin to be the default size
> 
> <img alt="Screen Shot 2021-12-23 at 13 32 45" width="248" src="https://user-images.githubusercontent.com/40663133/147242265-52e41381-9f81-4798-8aa1-6683ffdb1ee9.png">

This "margin" stems from the fact that there's an extra panel body in there for additional route info. The condition to check this would be unnecessarily long... 

![image](https://user-images.githubusercontent.com/19704958/147441112-6d08b162-d582-4a5b-9367-b0da27b6cd2d.png)

Reply:
we can remove the double margin for the matches panel (or wherever the margin comes from) and add it for the additional info panel, so there is no complex condition.

Reply:
but we're using generic styling from fundamental components for this...